File: flutter/lib/domain/use_case/deck/get_all_decks_use_case.dart
import '../../entities/deck.dart';
import '../../repository/deck_repository.dart';

class GetAllDecksUseCase {
  final DeckRepository _deckRepository;

  GetAllDecksUseCase(this._deckRepository);

  Future<List<Deck>> call() async {
    return await _deckRepository.getAllDecksWithFlashcards();
  }
}

File: android/app/src/main/java/com/synergyboat/flashcardAi/domain/usecase/GetAllDecksUseCase.kt
package com.synergyboat.flashcardAi.domain.usecase

import com.synergyboat.flashcardAi.domain.entities.Deck
import com.synergyboat.flashcardAi.domain.repository.DeckRepository

class GetAllDecksUseCase(
    private val repository: DeckRepository
) {
    suspend operator fun invoke(): List<Deck> {
        return repository.getAllDecksWithFlashcards()
    }
}